internal/main.go

```diff
@@ -78,18 +207,63 @@ func RunTests(opts RunTestsOpts) error {
 	testArgs = sliceAppendIf(opts.FlagVerbose, testArgs, "-v")
 	testArgs = sliceAppendIf(!opts.FlagCache, testArgs, "-count=1")
 	testArgs = sliceAppendIf(opts.FlagCover, testArgs, "-cover")
-	testArgs = sliceAppendIf(opts.FlagCoverProfile != "" || opts.FlagCoverReport, testArgs, "-coverprofile="+coverProfile)
+	if coverProfile != "" {
+		testArgs = append(testArgs, "-coverprofile="+coverProfile)
+	}
```

Comment:
For consistency with the rest: 
`testArgs = sliceAppendIf(coverProfile != "", testArgs, "-coverprofile="+coverProfile)`